modules/islandora_scholar_embargo/includes/batch.inc

```diff
@@ -81,11 +92,19 @@ function islandora_scholar_embargo_lift_batch_function($pid, $ds, $query_array,
     $context['sandbox']['items'] = count($context['sandbox']['pids']);
     $context['results']['redirect'] = $pid;
     $context['results']['success'] = array();
+    $context['results']['skip'] = array();
   }
   $targetpid = array_pop($context['sandbox']['pids']);
   $context['sandbox']['progress']++;
-  islandora_scholar_embargo_lift_embargo($targetpid, $ds);
-  $context['results']['success'][] = $targetpid;
+  $allowed_cmodels = variable_get('islandora_embargo_content_models', array('ir:citationCModel', 'ir:thesisCModel'));
+  $object = islandora_object_load($targetpid);
+  if (count(array_intersect($allowed_cmodels, $object->models)) > 0) {
+    islandora_scholar_embargo_lift_embargo($targetpid, $ds);
+    $context['results']['success'][] = $targetpid;
```

Comment:
This pid business can get very large and break the batch; suggest going with a count.